### PR TITLE
[DNM] server: handle test request in the batch commands

### PR DESCRIPTION
**Pull request by benpigchu**
## What have you changed? (mandatory)

Handle the batch test message type in order to set up a fuzz test around message batching.
See pingcap/kvproto#384

## What are the type of the changes? (mandatory)

Misc (other changes)

## How has this PR been tested? (mandatory)

`make dev`

## Does this PR affect documentation (docs) or release note? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

### CODE REVIEW

File: src/server/service/kv.rs [hunk 1]
```diff
@@ -1125,6 +1125,16 @@ fn handle_batch_commands_request<E: Engine>(
                 .map_err(|_| GRPC_MSG_FAIL_COUNTER.coprocessor.inc());
             response_batch_commands_request(id, resp, tx, timer);
         }
+        Some(BatchCommandsRequest_Request_oneof_cmd::Test(_req)) => {
```
Comment (hicqu) [inline src/server/service/kv.rs]: Is it ok to put the feature gate before this line?

File: Cargo.toml [hunk 2]
```diff
@@ -16,6 +16,7 @@ portable = ["engine/portable"]
 sse = ["engine/sse"]
 mem-profiling = ["tikv_alloc/mem-profiling"]
 no-fail = ["fail/no_fail"]
+batch-test = []
```
Comment (hicqu) [inline Cargo.toml]: Please rename the feature to service-test.

File: Cargo.toml [hunk 3]
```diff
@@ -135,6 +136,9 @@ features = ["nightly", "push", "process"]
 version = "0.1.4"
 default-features = false
 
+[patch."https://github.com/pingcap/kvproto.git"]
```
Comment (hicqu) [inline Cargo.toml]: Please make the patch under the "service-test" feature.

File: src/server/service/kv.rs (hunk 1 above)
Comment (hicqu) [inline src/server/service/kv.rs]: s/_req/req/

File: Cargo.toml [hunk 4]
```diff
@@ -135,6 +136,9 @@ features = ["nightly", "push", "process"]
...
+kvproto = {git = "https://github.com/benpigchu/kvproto.git", branch = "batch-test"}
```
Comment (hicqu) [inline Cargo.toml]: And plese use the repo `pingcap/kvproto.git`, branch `tikv-service-test`.

File: Cargo.toml (hunk 3 above)
Comment (benpigchu) [reply Cargo.toml]: You can't put patch under a feature

**Comment by overvenus** [comment]
PTAL @benpigchu  , should we close this PR?

**Comment by overvenus** [comment]
This PR is being closed because it is inactive.

**Pull request closed without merging**